site2/tools/build-site.sh

```diff
@@ -66,13 +66,14 @@ cd ${ROOT_DIR}/site2/website
 yarn
 yarn write-translations
 
-if [ "$CROWDIN_DOCUSAURUS_API_KEY" != "UNSET" ]; then
+# The crowdin upload and download take a long time to run, and have resulted in timeouts. In order to ensure that the
+# website is still able to get published, we only run the download and upload if current hour is 0-5.
+# This leads to executing crowdin-upload and crowdin-download once per day when website build is scheduled
+# to run with cron expression '0 */6 * * *'
+CURRENT_HOUR=$(date +%H)
+if [[ "$CROWDIN_DOCUSAURUS_API_KEY" != "UNSET" || $CURRENT_HOUR -lt 6 ]]; then
   # upload only if environment variable CROWDIN_UPLOAD=1 is set
-  # or current hour is 0-5
-  # this leads to executing crowdin-upload once per day when website build is scheduled
-  # to run with cron expression '0 */6 * * *'
-  CURRENT_HOUR=$(date +%H)
-  if [[ "$CROWDIN_UPLOAD" == "1" || $CURRENT_HOUR -lt 6 ]]; then
+  if [[ "$CROWDIN_UPLOAD" == "1" ]]; then
     yarn run crowdin-upload
   fi
   yarn run crowdin-download
```

Comment:
I'm fine in making the change that you have proposed.

I'm just thinking that when both crowdin-upload and crowdin-download will happen once a day, that might be so slow that it never completes. For context, I made the original `$CURRENT_HOUR -lt 6` hack in https://github.com/apache/pulsar/pull/10794 . That was a quick fix to get the website build to complete by running crowdin-upload once per day. 
One possible solution here would be to schedule crowdin-upload and crowdin-download in different time slots. 
However, it would be better to get them scheduled in completely separate jobs to fix the problem. That might be the only way to get the website build stable.

Reply:
I think the long-term fix is to upgrade the crowdin cli. Their docs mention that it is more efficient and has multi-threading for uploads.

I agree that separate jobs would make sense too. It only makes sense to upload files when docs have actually changed. I wonder if we can upload _just_ the changed files.